File: pocket-ec/src/main/java/com/dididi/pocket/ec/main/mall/goods/GoodsListDelegate.java
package com.dididi.pocket.ec.main.mall.goods;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.view.ViewPager;
import android.view.View;

import com.dididi.pocket.ec.R;
import com.dididi.pocket.ec.R2;
import com.dididi.pocket.core.ui.item.SearchBarItem;
import com.dididi.pocket.core.delegates.PocketDelegate;
import com.dididi.pocket.ec.main.mall.goods.fragment.AllGoodsList;
import com.dididi.pocket.ec.main.mall.goods.fragment.BestGoodsList;
import com.dididi.pocket.ec.main.mall.goods.fragment.MerchantGoodsList;
import com.ogaclejapan.smarttablayout.SmartTabLayout;
import com.ogaclejapan.smarttablayout.utils.v4.FragmentPagerItemAdapter;
import com.ogaclejapan.smarttablayout.utils.v4.FragmentPagerItems;

import butterknife.BindView;


/**
 * @describe 商品页面
 * @author dididi
 * @since 05/08/2018 .
 */

public class GoodsListDelegate extends PocketDelegate
        implements View.OnClickListener {

    @BindView(R2.id.delegate_mall_goods_viewpager)
    ViewPager mViewPager = null;
    @BindView(R2.id.delegate_mall_goods_searchBar)
    SearchBarItem mSearchBar = null;
    @BindView(R2.id.delegate_mall_goods_chooseBar)
    SmartTabLayout mChooseBar = null;

    @Override
    public Object setLayout() {
        return R.layout.delegate_mall_goods_list;
    }

    @Override
    public void onBindView(@Nullable Bundle savedInstanceState, View rootView) {
        FragmentPagerItemAdapter adapter =
                new FragmentPagerItemAdapter(getChildFragmentManager(),FragmentPagerItems.with(getContext())
                        .add(R.string.all,AllGoodsList.class)
                        .add(R.string.best,BestGoodsList.class)
                        .add(R.string.store,MerchantGoodsList.class)
                        .create());
        mViewPager.setAdapter(adapter);
        mChooseBar.setViewPager(mViewPager);
        mSearchBar.setLeftIcon("{faw-chevron-left}");
        mSearchBar.setLeftIconListener(this);
    }

    /**
     * 设置侧滑返回
     *
     * @param enable true即为可以侧滑返回
     */
    @Override
    public void setSwipeBackEnable(boolean enable) {
        super.setSwipeBackEnable(true);
    }

    @Override
    public void onClick(View view) {
        if (view.getId() == mSearchBar.getLeftIconId()) {
            getSupportDelegate().pop();
        }
    }

}

File: pocket-ec/src/main/java/com/dididi/pocket/ec/main/mall/goods/GoodsListDelegate.kt
package com.dididi.pocket.ec.main.mall.goods

import android.os.Bundle
import android.view.View
import com.dididi.pocket.core.delegates.PocketDelegate
import com.dididi.pocket.ec.R
import com.dididi.pocket.ec.main.mall.goods.fragment.AllGoodsList
import com.dididi.pocket.ec.main.mall.goods.fragment.BestGoodsList
import com.dididi.pocket.ec.main.mall.goods.fragment.MerchantGoodsList
import com.gyf.immersionbar.ktx.immersionBar
import com.ogaclejapan.smarttablayout.utils.v4.FragmentPagerItemAdapter
import com.ogaclejapan.smarttablayout.utils.v4.FragmentPagerItems
import kotlinx.android.synthetic.main.delegate_mall_goods_list.*


/**
 * @describe 商品页面
 * @author dididi
 * @since 05/08/2018 .
 */

class GoodsListDelegate : PocketDelegate(), View.OnClickListener {

    override fun setLayout(): Any {
        return R.layout.delegate_mall_goods_list
    }

    override fun onBindChildView(savedInstanceState: Bundle?, rootView: View?) {
    }

    override fun onBindView(savedInstanceState: Bundle?, rootView: View?) {
        val adapter = FragmentPagerItemAdapter(childFragmentManager, FragmentPagerItems.with(context)
                .add(R.string.all, AllGoodsList::class.java)
                .add(R.string.best, BestGoodsList::class.java)
                .add(R.string.store, MerchantGoodsList::class.java)
                .create())
        delegate_mall_goods_viewpager!!.adapter = adapter
        delegate_mall_goods_chooseBar!!.setViewPager(delegate_mall_goods_viewpager)
        delegate_mall_goods_searchBar!!.setLeftIcon(context?.getString(R.string.faw_chevron_left))
        delegate_mall_goods_searchBar!!.setLeftIconListener(this)
    }

    override fun getTitleBarId() = R.id.delegate_mall_goods_toolbar

    override fun initImmersionBar() {
        immersionBar {
            flymeOSStatusBarFontColor(R.color.textColorWhite)
            keyboardEnable(true)
        }
    }

    /**
     * 设置侧滑返回
     *
     * @param enable true即为可以侧滑返回
     */
    override fun setSwipeBackEnable(enable: Boolean) {
        super.setSwipeBackEnable(true)
    }

    override fun onClick(view: View) {
        if (view.id == delegate_mall_goods_searchBar!!.leftIconId) {
            supportDelegate.pop()
        }
    }

}
